src/components/fx/hover.js

```diff
@@ -203,6 +205,8 @@ function _hover(gd, evt, subplot, noHoverEvent) {
 
     var hovermode = evt.hovermode || fullLayout.hovermode;
 
+    if(hovermode && !supportsCompare) hovermode = 'closest';
```

Comment:
The second part of #2338 - fixes hover labels for geo and ternary on the `plot_types` mock

Reply:
Looks good. Not sure what's the best to :lock: this down?

Maybe just checking that `hovermode: 'x'` behaves the same as `hovermode: 'closest'` in a cartesian + geo (or ternary or polar) graph?

Reply:
good call -> https://github.com/plotly/plotly.js/pull/2339/commits/16b01f002b4b2874ff191bde6ce9cc9f9f38f1db